api/src/controllers/document-controller.js

```diff
@@ -254,13 +263,153 @@ async function createDocument(fileName, addedBy, url = null) {
   document.addedBy = addedBy;
   document.url = url;
   document.key = `${document._id}/${fileName}`;
-  document.read = ['public', 'sysadmin'];
+  document.read = ['sysadmin'];
   document.write = ['sysadmin'];
+
+  if (!isAnonymous) {
+    document.read.push('public');
+  }
+
   return await document.save();
 }
 
 exports.deleteS3Document = deleteS3Document;
 
 async function deleteS3Document(docKey) {
   return await s3.deleteObject({ Bucket: process.env.OBJECT_STORE_bucket_name, Key: docKey }).promise();
+}
+
+exports.uploadS3Document = uploadS3Document;
+
+async function uploadS3Document(docKey, docBody) {
+  // TODO: ACL is set to public-read until we know publishing rules.
```

Comment:
Is this a real TODO?

Reply:
Not sure, would have to ask @danieltruong 

I only moved this to a function, but didn't touch any of the S3 related logic (I don't have any experience with S3, or any of nrpti's S3 code).